src/extensions/storage_macro/dbc.rs

```diff
@@ -0,0 +1,58 @@
+use async_recursion::async_recursion;
+use rdbc2;
+
+type Error = Box<dyn std::error::Error + Send + Sync + 'static>;
+
+impl crate::application::CoLink {
+    #[async_recursion]
+    async fn _search_and_generate_query_string(
+        &self,
+        string_before_dbc: &str,
+        string_after_dbc: &str,
+    ) -> Result<String, Error> {
+        let split_key_path: Vec<&str> = string_after_dbc.split(':').collect();
+        println!("split_key_path: {:?}", split_key_path);
+        for i in 0..split_key_path.len() {
+            let current_key_path = format!(
+                "{}:{}",
+                string_before_dbc,
+                split_key_path[0..i + 1].join(":")
+            );
+            println!("current_key_path: {}", current_key_path);
+            let payload = self.read_entry(current_key_path.as_str()).await;
+            if payload.is_ok() {
+                let query_string_raw = String::from_utf8(payload.unwrap())?;
+                let count = query_string_raw.matches('?').count();
+                if count != split_key_path.len() - (i + 1) {
+                    return Err("Number of parameters does not match specified query string")?;
+                }
+                let mut query_string = query_string_raw;
+                for j in 0..count {
+                    query_string = query_string.replacen('?', split_key_path[i + j + 1], 1);
```

Comment:
wait... I thought this is handled inside rdbc2?

Reply:
Yes, but I have to loop over the addresses to check where the query is stored. So rather than using two loops (one here and one in rdbc2) I just do it in one here. I can also get the params out from the loop and use rdbc2 if you want